evolver/app/tests/test_calibration.py

```diff
@@ -153,13 +113,197 @@ def test_dispatch_temperature_calibration_action():
     }
 
     # Test the "temp" hardware's calibrator initialization
-    response = client.post("/hardware/temp/calibrator/start", json=request_payload)
+    response = client.post("/hardware/temp/calibrator/procedure/start", json=request_payload)
     assert response.status_code == 200
 
     # Now we will dispatch an action to the calibrator
     action_payload = {"action_name": "Vial_0_Temp_Reference_Value_Action", "payload": {"temperature": 25.0}}
 
     # Dispatch the action
-    dispatch_response = client.post("/hardware/temp/calibrator/dispatch", json=action_payload)
+    dispatch_response = client.post("/hardware/temp/calibrator/procedure/dispatch", json=action_payload)
+    assert dispatch_response.status_code == 200
+    assert dispatch_response.json() == {
+        "state": {"selected_vials": [0, 1, 2], "temp": {"vial_0": {"reference": [25.0], "raw": []}}}
+    }
+
+
+def test_dispatch_temperature_calibration_raw_value_action():
+    # Set up the evolver instance with hardware and a Temperature Calibrator
+    temp_calibrator = TemperatureCalibrator()  # Ensure this is properly initialized
+
+    # Create NoOpSensorDriver and assign the temp calibrator to it
+    evolver_instance = Evolver(
+        hardware={"temp": NoOpSensorDriver(name="temp", calibrator=temp_calibrator, vials=[0, 1, 2])}
+    )
+
+    # Ensure the temp calibrator has access to the evolver
+    temp_calibrator.evolver = evolver_instance
+
+    # Set the evolver state in the app before testing
+    app.state.evolver = evolver_instance
+
+    # Mock the hardware read method to return a meaningful value
+    evolver_instance.hardware["temp"].read = lambda: [1.23, 2.34, 3.45]
+
+    # Create the test client
+    client = TestClient(app)
+
+    # Prepare the request payload to initialize the calibration procedure
+    request_payload = {
+        "selected_vials": [0, 1, 2]  # Simulate the user selecting vials for calibration
+    }
+
+    # Test the "temp" hardware's calibrator initialization
+    response = client.post("/hardware/temp/calibrator/procedure/start", json=request_payload)
+    assert response.status_code == 200
+
+    # Now we will dispatch an action to the calibrator
+    action_payload = {"action_name": "Vial_0_Temp_Raw_Voltage_Action", "payload": {}}
+
+    # Dispatch the action
+    dispatch_response = client.post("/hardware/temp/calibrator/procedure/dispatch", json=action_payload)
     assert dispatch_response.status_code == 200
-    assert dispatch_response.json() == {"state": {"temp": {"vial_0": {"reference": [25.0], "raw": []}}}}
+    assert dispatch_response.json() == {
+        "state": {"selected_vials": [0, 1, 2], "temp": {"vial_0": {"reference": [], "raw": [1.23]}}}
+    }
+
+
+def test_dispatch_temperature_calibration_calculate_fit_action():
```

Comment:
New test where a fit (fit method is on the Calibrator's input_/output_transformer) is calculated on the measured_data (reference and raw)